src/Microsoft.DotNet.ProjectModel/Graph/LockFile.cs

```diff
@@ -87,5 +84,18 @@ public bool IsValidForProject(Project project, out string message)
             message = null;
             return true;
         }
+
+        public void MergeWith(LockFile fragmentLockFile)
+        {
+            if (Version != fragmentLockFile.Version)
+            {
+                throw new ArgumentException($"Cannot merge master lock file {LockFilePath} with fragment lock file {fragmentLockFile.LockFilePath} due to missmatching versions");
+            }
+
+            Targets.MergeWith(fragmentLockFile.Targets, l => l.TargetFramework.Framework + l.RuntimeIdentifier);
```

Comment:
I see now that this is just for de-duping, but still, I'd add a separator just in case, and you may as well use the one used in the lock file.